src/completion_queue.rs

```diff
@@ -59,22 +82,18 @@ impl CompletionQueue {
         Ok(())
     }
 
-    /// Poll `num_entries` work completions from CQ
-    pub(crate) fn poll(&self, num_entries: usize) -> io::Result<Vec<WorkCompletion>> {
-        let mut ans: Vec<WorkCompletion> = Vec::with_capacity(num_entries);
-        let poll_res =
-            unsafe { ibv_poll_cq(self.as_ptr(), num_entries.cast(), ans.as_mut_ptr().cast()) };
-        if poll_res >= 0_i32 {
-            assert!(num_entries >= poll_res.cast());
-            let poll_res = poll_res.cast();
-
-            // the length equals to the poll results length
-            unsafe { ans.set_len(poll_res) };
-            ans.shrink_to(poll_res);
-
-            assert_eq!(ans.len(), poll_res);
-            assert_eq!(ans.capacity(), poll_res);
-            Ok(ans)
+    /// Poll work completions from CQ up to `DEFAULT_CQ_SIZE` at a time
+    pub(crate) fn poll_cq_multiple(&self, wc_buf: &mut Vec<WorkCompletion>) -> io::Result<()> {
+        let cqe_num =
```

Comment:
It's necessary to check the buffer capacity before calling ibv_poll_cq. Otherwise the fn poll_cq_multiple should be unsafe fn.

Reply:
Is there anything that will change the cap? Or just ensure the safety requirements of `set_len`.

Reply:
Just ensure the safety requirements.